macros/edr/materializations/tests/test.sql

```diff
@@ -0,0 +1,238 @@
+  {% set full_table_name = elementary.insensitive_get_dict_value(elementary_test_row, 'full_table_name') %}
+  {% set database_name, schema_name, table_name = elementary.split_full_table_name_to_vars(full_table_name) %}
+  {% set test_params = elementary.insensitive_get_dict_value(flattened_test, 'test_params', {}) %}
+  {% set sensitivity = elementary.insensitive_get_dict_value(test_params, 'sensitivity') or elementary.get_config_var('anomaly_sensitivity') %}
+  {% set backfill_days = elementary.insensitive_get_dict_value(test_params, 'backfill_days') or elementary.get_config_var('backfill_days') %}
+  {% set timestamp_column = elementary.insensitive_get_dict_value(test_params, 'timestamp_column') %}
+  {% set parent_model_unique_id = elementary.insensitive_get_dict_value(flattened_test, 'parent_model_unique_id') %}
+  {% if not timestamp_column %}
+    {% set parent_model_node = elementary.get_node(parent_model_unique_id) %}
+    {% set timestamp_column = elementary.get_timestamp_column(timestamp_column, parent_model_node) %}
+  {% endif %}
+  {% do test_params.update({'sensitivity': sensitivity, 'timestamp_column': timestamp_column, 'backfill_days': backfill_days}) %}
+  {% set column_name = elementary.insensitive_get_dict_value(elementary_test_row, 'column_name') %}
+  {% set metric_name = elementary.insensitive_get_dict_value(elementary_test_row, 'metric_name') %}
+  {% set metric_id = elementary.insensitive_get_dict_value(elementary_test_row, 'metric_id') %}
+  {% set backfill_period = "'-" ~ backfill_days ~ "'" %}
+  {% set test_results_query %}
+      with anomaly_scores as (
+          select * from {{ elementary.get_elementary_test_table(flattened_test.name, 'anomaly_scores') }}
+      ),
+      anomaly_scores_with_is_anomalous as (
+      select  *,
+              case when abs(anomaly_score) > {{ sensitivity }}
+              and bucket_end >= {{ elementary.timeadd('day', backfill_period, elementary.get_max_bucket_end()) }}
+              and training_set_size >= {{ elementary.get_config_var('days_back') -1 }} then TRUE else FALSE end as is_anomalous
+          from anomaly_scores
+          where anomaly_score is not null
+      )
+      select metric_value as value,
+             training_avg as average,
+             case when is_anomalous = TRUE then
+              lag(min_metric_value) over (partition by full_table_name, column_name, metric_name, dimension, dimension_value order by bucket_end)
+             else min_metric_value end as min_value,
+             case when is_anomalous = TRUE then
+              lag(max_metric_value) over (partition by full_table_name, column_name, metric_name, dimension, dimension_value order by bucket_end)
+              else max_metric_value end as max_value,
+             bucket_start as start_time,
+             bucket_end as end_time,
+             is_anomalous,
+             dimension,
+             dimension_value,
```

Comment:
Where are you using the added `dimension` and `dimension_value` fields? (didn't see them in `handle_test_result`)